### DISCUSSION TITLE
File Management - Blob Storage

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#1975] [General] [answered]
How do I extend File Management with Blob storage for MSSQL DB?

Upload very large file nearly 20-100 GB into Blob (or any recommended FM) and reading file data from database without any performance issues.

Current File Management hasn't any security risks?

Image, video, PDF, doc, etc., files in database will be difficult to search file contents? What is the best approach for storing files?

**Comment by sbwalker** [comment, marked as answer]
The current file manager only supports files saved locally to disk - they can be saved in either a public folder or secure private folder.  There is currently no support for saving files in a database or cloud service. if you need this functionality you will need to build it yourself. Note that Oqtane supports multiple databases - SQL Server, PostgreSQL, SQLite, mySQL and blob supports varies greatly across these databases so this could not be easily included as a core feature.

**Reply by Sharafudeen** [reply]
In some use case, users documents such passport, Identity cards, etc., won't be allowed to store as raw files in the server for security reasons. For this, if Oqtane can provide an option "File type = Blob or File System" in the framework level that would be great.

**Reply by sbwalker** [reply]
@Sharafudeen at some point in the future an extensibility model may be introduced in the file/folder management service but it is not as high of a priority as other items at this time. However you can certainly implement your own service for saving documents in a database or cloud service in your own custom modules.
